packages/dds/map/src/directory.ts

```diff
@@ -2181,18 +2185,21 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 	): boolean {
 		if (this.pendingClearMessageIds.length > 0) {
 			if (local) {
+				// Remove all pendingMessageIds lower than first pendingClearMessageId.
+				// Non null asserting, because of pendingClearMessageIds length check above
+				// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+				const firstPendingClearMessageId = this.pendingClearMessageIds[0]!;
 				assert(
 					localOpMetadata !== undefined &&
 						isKeyEditLocalOpMetadata(localOpMetadata) &&
-						localOpMetadata.pendingMessageId < this.pendingClearMessageIds[0],
+						localOpMetadata.pendingMessageId < firstPendingClearMessageId,
 					0x010 /* "Received out of order storage op when there is an unackd clear message" */,
 				);
-				// Remove all pendingMessageIds lower than first pendingClearMessageId.
-				const lowestPendingClearMessageId = this.pendingClearMessageIds[0];
 				const pendingKeyMessageIdArray = this.pendingKeys.get(op.key);
 				if (pendingKeyMessageIdArray !== undefined) {
 					let index = 0;
-					while (pendingKeyMessageIdArray[index] < lowestPendingClearMessageId) {
+					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
```suggestion
					// Non-null asserting because we maintain that the pendingKeyMessageIdArray will only exist if it is non-empty.
					// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```